apps/studio/src/features/editing-experience/components/utils.ts

```diff
@@ -74,3 +78,16 @@ export const uploadModifiedAssets = async ({
     return true
   })
 }
+
+export const generatePreviewSitemap = (
+  sitemap: typeof collectionSitemap,
+  title = "Your filename",
+) => {
+  return {
+    ...sitemap,
+    children: sitemap.children.map(({ children, ...rest }) => ({
+      ...rest,
+      children: children.map((props) => ({ ...props, title })),
+    })),
+  } as IsomerGeneratedSiteProps["siteMap"]
```

Comment:
nit: can this be defined as part of the function return type?

Reply:
it needs the cast because the json tyhpe cannot be infered properly - it's inferred as `string` rather than the most specific but becaus eit's a readonly constant, we know it's correct (hence the cast)